packages/editor-ui/src/plugins/codemirror/completions/infoBoxRenderer.ts

```diff
@@ -0,0 +1,228 @@
+		returnTypeSpan.textContent = ': ' + doc.returnType;
+
+		header.appendChild(propNameSpan);
+		header.appendChild(returnTypeSpan);
+	}
+	return header;
+};
+
+const renderDescription = ({
+	description,
+	docUrl,
+	example,
+}: {
+	description: string;
+	docUrl?: string;
+	example?: DocMetadataExample;
+}): HTMLElement => {
+	const descriptionBody = document.createElement('div');
+	descriptionBody.classList.add('autocomplete-info-description');
+	const descriptionText = document.createElement('p');
+	descriptionText.innerHTML = sanitizeHtml(
+		description.replace(/`(.*?)`/g, '<code>$1</code>') + ' ',
+	);
+	descriptionBody.appendChild(descriptionText);
+
+	if (docUrl) {
+		const descriptionLink = document.createElement('a');
+		descriptionLink.setAttribute('target', '_blank');
+		descriptionLink.setAttribute('href', docUrl);
+		descriptionLink.innerText =
+			i18n.autocompleteUIValues.docLinkLabel ?? i18n.baseText('generic.learnMore');
+		descriptionLink.addEventListener('mousedown', (event: MouseEvent) => {
+			// This will prevent documentation popup closing before click
+			// event gets to links
+			event.preventDefault();
+		});
+		descriptionLink.classList.add('autocomplete-info-doc-link');
+		descriptionText.appendChild(descriptionLink);
+	}
+
+	if (example) {
+		const renderedExample = renderExample(example);
+		descriptionBody.appendChild(renderedExample);
+	}
+
+	return descriptionBody;
+};
+
+const renderArgs = (args: DocMetadataArgument[]): HTMLElement => {
+	const argsContainer = document.createElement('div');
+	argsContainer.classList.add('autocomplete-info-args-container');
+
+	const argsTitle = document.createElement('div');
+	argsTitle.classList.add('autocomplete-info-section-title');
+	argsTitle.textContent = i18n.baseText('codeNodeEditor.parameters');
+	argsContainer.appendChild(argsTitle);
+
+	const argsList = document.createElement('ul');
+	argsList.classList.add('autocomplete-info-args');
+
+	for (const arg of args) {
+		const argItem = document.createElement('li');
+		const argName = document.createElement('span');
+		argName.classList.add('autocomplete-info-arg-name');
+		argName.textContent = arg.name;
+		if (arg.optional === true) argName.textContent += '?';
+
+		argItem.appendChild(argName);
+
+		if (arg.type) {
+			const argType = document.createElement('span');
+			argType.classList.add('autocomplete-info-arg-type');
+			argType.textContent = `: ${arg.type}`;
+			argItem.appendChild(argType);
+		}
+
+		if (arg.description) {
+			const argDescription = document.createElement('span');
+			argDescription.classList.add('autocomplete-info-arg-description');
+			argDescription.innerHTML = `- ${sanitizeHtml(
+				arg.description.replace(/`(.*?)`/g, '<code>$1</code>'),
+			)}`;
+
+			argItem.appendChild(argDescription);
+		}
+
+		argsList.appendChild(argItem);
+	}
+
+	argsContainer.appendChild(argsList);
+	return argsContainer;
+};
+
+const renderExample = (example: DocMetadataExample): HTMLElement => {
```

Comment:
Nit - Return type is inferrable - also the inferred type is narrower than this. Same for others in this file.